sdk/src/main/scala/com/horizen/SidechainStateUtxoMerkleTreeProvider.scala

```diff
@@ -0,0 +1,70 @@
+package com.horizen
+
+import com.horizen.box.CoinsBox
+import com.horizen.proposition.PublicKey25519Proposition
+import com.horizen.storage.{SidechainStateUtxoMerkleTreeStorage, SidechainStorageInfo}
+import com.horizen.utils.ByteArrayWrapper
+
+import scala.util.Try
+
+trait SidechainStateUtxoMerkleTreeProvider extends SidechainStorageInfo {
+  def rollback(version: ByteArrayWrapper): Try[SidechainStateUtxoMerkleTreeProvider]
+
+  def getMerklePath(boxId: Array[Byte]): Option[Array[Byte]]
+
+  def update(version: ByteArrayWrapper,
+             boxesToAppend: Seq[SidechainTypes#SCB],
+             boxesToRemoveSet: Set[ByteArrayWrapper]): Try[SidechainStateUtxoMerkleTreeProvider]
+
+  def getMerkleTreeRoot: Array[Byte]
```

Comment:
Maybe we can set return type to `Option[Array[Byte]]` so `SidechainUtxoMerkleTreeProviderCSWDisabled` will return None instead of empty array.
See the usage of `getMerkleTreeRoot` whole committing the changes in the State.